### PR TITLE
8253208: Move CDS related code to a separate class

### PR DESCRIPTION
**Pull request by yminqi**
With more CDS related code added to VM, it is time to move CDS code to a separate class. CDS is the new class which is specific to CDS.

Tests: tier1-4

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253208](https://bugs.openjdk.java.net/browse/JDK-8253208): Move CDS related code to a separate class

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to c01deec7a056a2146b1107b4bfc545c9317b5e96

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/261/head:pull/261`
`$ git checkout pull/261`

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/misc/CDS.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     *
+     * Those static fields remain as uninitialized if there is no mapped CDS
+     * java heap data or there is any error during initialization of the
+     * object class in the archived graph.
+     */
+    public static native void initializeFromArchive(Class<?> c);
+
+    public static native void defineArchivedModules(ClassLoader platformLoader, ClassLoader systemLoader);
+
+    public static native long getRandomSeedForCDSDump();
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/misc/CDS.java:42]: The moving of the archive methods to CDS looks okay but inconsistent to only comment 3 of the 5 methods.

File: src/java.base/share/classes/jdk/internal/misc/CDS.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Check if CDS dynamic dumping is enabled via the DynamicDumpSharedSpaces flag.
+     */
+    public static native boolean isCDSDumpingEnabled();
+
+    /**
+     * Check if CDS sharing is enabled by via the UseSharedSpaces flag.
+     */
+    public static native boolean isCDSSharingEnabled();
```
Comment (iklam) [inline src/java.base/share/classes/jdk/internal/misc/CDS.java:52]: I think the word CDS is redundant in the method names. How about

getRandomSeedForCDSDump() -> getRandomSeedForDumping()
isCDSDumpingEnabled() -> isDynamicDumpingEnabled()  // doesn't return true if we're doing a static dump
isCDSSharingEnabled() -> isSharingEnabled()

File: src/java.base/share/native/libjava/CDS.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+JNIEXPORT jlong JNICALL
+Java_jdk_internal_misc_CDS_getRandomSeedForCDSDump(JNIEnv *env, jclass ignore) {
+    return JVM_GetRandomSeedForCDSDump();
+}
+
+JNIEXPORT jboolean JNICALL
+Java_jdk_internal_misc_CDS_isCDSDumpingEnabled(JNIEnv *env, jclass jcls) {
+    return JVM_IsCDSDumpingEnabled(env);
```
Comment (iklam) [inline src/java.base/share/native/libjava/CDS.c:49]: Maybe: return JVM_IsCDSDynamicDumpingEnabled(env)

Comment (yminqi) [reply src/java.base/share/native/libjava/CDS.c]: updated with comments.

**Review by mlchung — COMMENTED** [review]
This patch looks okay.   Please add the javadoc for `CDS::getRandomSeedForDumping` and `CDS::defineArchiveModules` for completeness.

**Review by erikj79 — COMMENTED** [review]
Build changes look good.

**Review by iklam — APPROVED** [review]

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**